library/pool.c

```diff
@@ -55,7 +55,10 @@ static void dnet_work_pool_stop(struct dnet_work_pool_place *place)
 
 	for (i = 0; i < place->pool->num; ++i) {
 		wio = &place->pool->wio_list[i];
-		pthread_join(wio->tid, NULL);
+		if (!wio->joined) {
+			pthread_join(wio->tid, NULL);
+			wio->joined = 1;
+		}
```

Comment:
Actually, this fix logically connected with the fix in dnet_server_node_destroy().
Previously, backends array from dnet_io struct was not freed in dnet_server_node_destroy(). Also, dnet_node_cleanup_common_resources() was called before dnet_backend_cleanup_all().
dnet_backend_cleanup_all() cleans caches and syncs caches to backends.
Now, backends array is freed in dnet_node_cleanup_common_resources(). dnet_work_pool_stop() is called from dnet_node_stop_common_resources() and subsequently from dnet_backend_cleanup_all() leading to joining same thread twice, causing crash in libpthread. This is the answer why I have added this check.
Why dnet_work_pool_stop() was not called twice before? dnet_node_cleanup_common_resources() was called before dnet_backend_cleanup_all(), so cleanups from dnet_node_cleanup_common_resources() prevented second dnet_work_pool_stop() from dnet_backend_cleanup_all().